### PR TITLE
Safeguarding TransformerFactory from XXE

### PR DESCRIPTION
**Pull request by prasantakumar82**

### CODE REVIEW

**Review by GaryWKeim — APPROVED** [review]

File: dynamic-config/cli/upgrade-tools/src/main/java/org/terracotta/dynamic_config/cli/upgrade_tools/config_converter/xml/XmlUtility.java [hunk 1]
```diff
@@ -79,7 +80,9 @@ public static void removeNode(Node node, boolean removeEmptyParent) {
   }
 
   public static String getPrettyPrintableXmlString(Node doc) throws TransformerException {
-    Transformer transformer = TransformerFactory.newInstance().newTransformer();
+    TransformerFactory factory = TransformerFactory.newInstance();
+    factory.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
```
Comment (cljohnso) [inline dynamic-config/cli/upgrade-tools/src/main/java/org/terracotta/dynamic_config/cli/upgrade_tools/config_converter/xml/XmlUtility.java:84]: Should we also be using `ACCESS_EXTERNAL_DTD` and/or `ACCESS_EXTERNAL_SCHEMA` as suggested in to doc?

**Review by cljohnso — COMMENTED** [review]
Should we be using using more settings (noted elsewhere).

File: dynamic-config/cli/upgrade-tools/src/main/java/org/terracotta/dynamic_config/cli/upgrade_tools/config_converter/xml/XmlUtility.java (hunk 1 above)
Comment (prasantakumar82) [reply dynamic-config/cli/upgrade-tools/src/main/java/org/terracotta/dynamic_config/cli/upgrade_tools/config_converter/xml/XmlUtility.java:84]: No, it's not required if Feature for Secure Processing (FSP) is on. Those properties can be used for selectively permits/restricts external access.

**Comment by prasantakumar82** [comment]
> Should we be using using more settings (noted elsewhere).

Not required, turning on Feature for Secure Processing (FSP) is sufficient.

**Comment by GaryWKeim** [comment]
I'm unsure of the possible daggy requirements for terracotta-platform here.

**Pull request merged**